backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+func (d *dtmb) CreateTaskHub(ctx context.Context) error {
+	_, err := d.clientClient.Metadata(ctx, &dtmbprotos.MetadataRequest{})
+	if err != nil {
+		return backend.ErrTaskHubNotFound
+	}
+	return nil
+}
+
+// DeleteTaskHub deletes an existing task hub configured for the current backend. It's up to the backend
+// implementation to determine how the task hub data is deleted.
+//
+// If the task hub for this backend doesn't exist, an error of type [ErrTaskHubNotFound] is returned.
+func (d *dtmb) DeleteTaskHub(context.Context) error {
+	return fmt.Errorf("the TaskHub cannot be deleted via the API. Please delete the TaskHub manually")
+}
+
+// Start starts any background processing done by this backend.
+func (d *dtmb) Start(ctx context.Context, orchestrators *[]string, activities *[]string) error {
+	// TODO: is the context provided a background context?
+
+	ctxWithCancel, cancel := context.WithCancel(ctx)
+	d.workerCancelFunc = cancel
+
+	err := d.connectWorker(ctxWithCancel, orchestrators, activities)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *dtmb) connectWorker(ctx context.Context, orchestrators *[]string, activities *[]string) error {
+	// Establish the ConnectWorker stream
+	worker := d.workerClient
+	var taskHubName string = "MAKE THIS CONFIGURABLE"
+	var testID string = "MAKE THIS CONFIGURABLE"
+
+	readyCh := make(chan struct{})
+	bgErrCh := make(chan error)
+	d.connectWorkerClientStream = make(chan *dtmbprotos.ConnectWorkerClientMessage)
+	serverMessageChan := make(chan *dtmbprotos.ConnectWorkerServerMessage)
+
+	var orchestratorFnList utils.OrchestratorFnList = *orchestrators
+	var activityFnList utils.ActivityFnList = *activities
+
+	// start the bidirectional stream
+	go func() {
+		bgErrCh <- utils.ConnectWorker(
+			ctx,
+			testID,
+			taskHubName,
+			worker,
+			orchestratorFnList,
+			activityFnList,
+			serverMessageChan,
+			d.connectWorkerClientStream,
+			func() {
+				close(readyCh)
+			},
+			true,
+		)
+	}()
+
+	// Wait for readiness
+	select {
+	case err := <-bgErrCh: // This includes a timeout too
+		log.Printf("[%s] Error starting test: %v", testID, err)
+		return err
+	case <-readyCh:
+		// All good
+	}
+
+	// Do a ping as last warm-up
+	pingCtx, pingCancel := context.WithTimeout(ctx, 15*time.Second)
+	pingCtx = metadata.AppendToOutgoingContext(pingCtx,
+		"taskhub", taskHubName,
+	)
+	_, err := worker.Ping(pingCtx, &dtmbprotos.PingRequest{})
+	pingCancel()
+	if err != nil {
+		log.Printf("[%s] Ping error: %v", testID, err)
+		return fmt.Errorf("ping error: %w", err)
+	}
+
+	// Execute messages in background
+	go func() {
+		for msg := range serverMessageChan {
+			switch m := msg.Message.(type) {
+
+			case *dtmbprotos.ConnectWorkerServerMessage_ExecuteActivity:
+				go func() {
+					d.activityQueue.Enqueue(m.ExecuteActivity)
+				}()
+
+			case *dtmbprotos.ConnectWorkerServerMessage_ExecuteOrchestration:
+				go func() {
```

Comment:
Do we need to do this in a background goroutine you think?

Reply:
I'll remove for now and we can evaluate later.